packages/@react-aria/checkbox/src/useCheckbox.ts

```diff
@@ -39,7 +39,7 @@ export function useCheckbox(props: AriaCheckboxProps, state: ToggleState, inputR
     if (inputRef.current) {
       inputRef.current.indeterminate = isIndeterminate;
     }
-  }, [inputRef, isIndeterminate]);
+  });
```

Comment:
`isIndeterminate` is meant to be controlled. But, the browser resets it to false when clicking on an indeterminate checkbox. It only "works" because of the spectrum styling (but not correct semantically). If you went to the docs page for react-aria and clicked on the indeterminate checkbox, it would become checked. Changing this makes this useEffect run every render and ensure the indeterminate state matches the prop.